here-naksha-app-service/src/main/java/com/here/naksha/app/service/http/tasks/AbstractApiTask.java

```diff
@@ -70,93 +75,70 @@ protected AbstractApiTask(
     return verticle.sendErrorResponse(routingContext, XyzError.NOT_IMPLEMENTED, "Unsupported operation!");
   }
 
-  protected <R extends XyzFeature> @NotNull XyzResponse transformReadResultToXyzCollectionResponse(
-      final @Nullable Result rdResult, final @NotNull Class<R> type) {
-    if (rdResult == null) {
-      // return empty collection
-      logger.warn("Unexpected null result, returning empty collection.");
-      return verticle.sendXyzResponse(
-          routingContext, HttpResponseType.FEATURE_COLLECTION, new XyzFeatureCollection());
-    } else if (rdResult instanceof ErrorResult er) {
-      // In case of error, convert result to ErrorResponse
-      logger.error("Received error result {}", er);
-      return verticle.sendErrorResponse(routingContext, er.reason, er.message);
-    } else if (rdResult instanceof ReadResult<?> rr) {
-      // In case of success, convert result to success XyzResponse
-      final List<R> features = new ArrayList<>();
-      int cnt = 0;
-      for (final R feature : rr.withFeatureType(type)) {
-        features.add(feature);
-        if (++cnt >= 1000) {
-          break; // TODO : can be improved later (perhaps by accepting limit as an input)
-        }
-      }
-      rr.close();
-      final XyzFeatureCollection response = new XyzFeatureCollection().withFeatures(features);
-      return verticle.sendXyzResponse(routingContext, HttpResponseType.FEATURE_COLLECTION, response);
-    }
-    // unexpected result type
-    logger.error("Unexpected result type {}", rdResult.getClass().getSimpleName());
-    return verticle.sendErrorResponse(
-        routingContext,
-        XyzError.EXCEPTION,
-        "Unsupported result type : " + rdResult.getClass().getSimpleName());
-  }
-
   protected <R extends XyzFeature> @NotNull XyzResponse transformReadResultToXyzFeatureResponse(
       final @NotNull Result rdResult, final @NotNull Class<R> type) {
-    if (rdResult == null) {
+    return transformResultToXyzFeatureResponse(
+        rdResult,
+        type,
+        () -> verticle.sendErrorResponse(
+            routingContext, XyzError.NOT_FOUND, "The desired feature does not exist."));
+  }
+
+  protected <R extends XyzFeature> @NotNull XyzResponse transformWriteResultToXyzFeatureResponse(
```

Comment:
this and the transformReadResultToXyzFeatureResponse() are the same, shall we remove them both, and replace any usages with the underlying transformResultToXyzFeatureResponse() ?

Reply:
Good catch - actually writing should have different handler fos situation when no elements are returned - it should not be `NOT_FOUND`. The similarity between these two methods was the main reason I extracted the thiird major one that they use. 
I did not pay enough attention and forgot to align the write method so it would use differn `onNoElementsReturned` - that's exactly why we need Pull Requests and review :)

Reply:
done